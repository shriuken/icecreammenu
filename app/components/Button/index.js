/**
 *
 * Button
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/* eslint-disable react/prefer-stateless-function */
class Button extends React.PureComponent {
  render() {
    const StyledButton = styled.button`
      background-color: ${this.props.positive ? '#4caf50' : 'red'};
      border-radius: 3px;
      color: white;
      padding: ${this.props.positive ? '15px 32px' : '4px 24px'};
      text-align: center;
      text-decoration: none;
      font-size: 16px;
      margin: 4px 2px;
      cursor: pointer;
    `;
    return (
      <StyledButton positive={this.props.positive} onClick={this.props.onClick}>
        {this.props.text}
      </StyledButton>
    );
  }
}

Button.propTypes = {
  positive: PropTypes.bool,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
