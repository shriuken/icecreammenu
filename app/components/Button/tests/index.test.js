import React from 'react';
import { shallow /* mount */ } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import Button from '../index';

describe('<Button />', () => {
  it('renders positive', () => {
    const component = shallow(<Button positive onClick={jest.fn()} text="" />);
    expect(component).toMatchSnapshot();
  });
  it('renders negatively', () => {
    const component = shallow(
      <Button positive={false} onClick={jest.fn()} text="" />,
    );
    expect(component).toMatchSnapshot();
  });
  it('renders positive differently from negative', () => {
    const positive = shallow(<Button positive onClick={jest.fn()} text="" />);
    const negative = shallow(
      <Button positive={false} onClick={jest.fn()} text="" />,
    );
    expect(positive).not.toEqual(negative);
  });
  // test onClick event is called.
});
