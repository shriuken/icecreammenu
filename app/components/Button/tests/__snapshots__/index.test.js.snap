// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders negatively 1`] = `
<Button__StyledButton
  onClick={[MockFunction]}
  positive={false}
/>
`;

exports[`<Button /> renders positive 1`] = `
<Button__StyledButton
  onClick={[MockFunction]}
  positive={true}
/>
`;
